staticdnshostprovider.go

```diff
@@ -0,0 +1,111 @@
+package zk
+
+import (
+	"fmt"
+	"log/slog"
+	"math/rand"
+	"net"
+	"sync"
+)
+
+type hostPort struct {
+	host, port string
+}
+
+func (hp *hostPort) String() string {
+	return hp.host + ":" + hp.port
+}
+
+// StaticHostProvider is the default HostProvider, and replaces the now deprecated DNSHostProvider.
+// It will iterate through the ZK hosts on every call to Next, and return a random address selected
+// from the resolved addresses of the ZK host (if the host is already an IP, it will return that
+// directly). It is important to manually resolve and shuffle the addresses because the DNS record
+// that backs a host may rarely (or never) change, so repeated calls to connect to this host may
+// always connect to the same IP. This mode is the default mode, and matches the Java client's
+// implementation. Note that if the host cannot be resolved, Next will return it directly, instead of
+// an error. This will cause Dial to fail and the loop will move on to a new host. It is implemented
+// as a pound-for-pound copy of the standard Java client's equivalent:
+// https://github.com/linkedin/zookeeper/blob/629518b5ea2b26d88a9ec53d5a422afe9b12e452/zookeeper-server/src/main/java/org/apache/zookeeper/client/StaticHostProvider.java#L368
+type StaticHostProvider struct {
+	mu         sync.Mutex // Protects everything, so we can add asynchronous updates later.
+	servers    []hostPort
+	curr       int
```

Comment:
These indexes are also a little bit confusing, so good idea to add some documentation too.

Reply:
Updated